src/libfuncs/cast.rs

```diff
@@ -224,56 +198,44 @@ pub fn build_downcast<'ctx, 'this>(
             (result, dst_ty)
         };
 
-        let max_value = block
-            .append_operation(arith::constant(
-                context,
-                Attribute::parse(
-                    context,
-                    &format!(
-                        "{}: {}",
-                        info.to_range
-                            .intersection(&info.from_range)
-                            .ok_or_else(|| Error::SierraAssert(
-                                "range should always interesct".to_string()
-                            ))?
-                            .upper,
-                        compare_ty
-                    ),
-                )
-                .ok_or_else(|| {
-                    Error::Error("downcast: failed to make max value attribute".to_string())
-                })?,
-                location,
-            ))
-            .result(0)?
-            .into();
+        let max_val_parse = Attribute::parse(
+            context,
+            &format!(
+                "{}: {}",
+                info.to_range
+                    .intersection(&info.from_range)
+                    .ok_or_else(|| Error::SierraAssert(
+                        "range should always interesct".to_string()
+                    ))?
+                    .upper,
+                compare_ty
+            ),
+        )
+        .ok_or_else(|| Error::Error("downcast: failed to make max value attribute".to_string()))?;
+
+        let max_value =
+            block.const_int_from_type(context, location, max_value_parse, compare_ty)?;
```

Comment:
you dont need to use Attribute parse, const_int_from_type does it internally, you need to pass it the value, in this case its the `.upper` of the intersection, and the type: `compare_ty`